src/coreclr/tools/Common/Compiler/DependencyAnalysis/Target_Riscv64/TargetRegisterMap.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis.Riscv64
+{
+    /// <summary>
+    /// Maps logical registers to physical registers on a specified OS.
+    /// </summary>
+    public struct TargetRegisterMap
+    {
+        public readonly Register Arg0;
+        public readonly Register Arg1;
+        public readonly Register Arg2;
+        public readonly Register Arg3;
+        public readonly Register Arg4;
+        public readonly Register Arg5;
+        public readonly Register Arg6;
+        public readonly Register Arg7;
+        public readonly Register IntraProcedureCallScratch1;
+        public readonly Register Result;
+
+        public TargetRegisterMap(TargetOS os)
+        {
+            Arg0 = Register.X10;
+            Arg1 = Register.X11;
+            Arg2 = Register.X12;
+            Arg3 = Register.X13;
+            Arg4 = Register.X14;
+            Arg5 = Register.X15;
+            Arg6 = Register.X16;
+            Arg7 = Register.X17;
+            IntraProcedureCallScratch1 = Register.X28;
+            Result = Register.X10;
```

Comment:
If your enum Register would be compatible with ABI, then here you wouldn't have *less readable* `Register.Xn`, but instead:
```C#
Arg0 = Register.A0;
Arg1 = Register.A1;
Arg2 = Register.A2;
Arg3 = Register.A3;
Arg4 = Register.A4;
Arg5 = Register.A5;
Arg6 = Register.A6;
Arg7 = Register.A7;
IntraProcedureCallScratch1 = Register.T3;
Result = Register.A0;
```